MediaContentSynchronization/Model/RemoveObsoleteContentAsset.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaContentSynchronization\Model;
+
+use Magento\MediaGallerySynchronizationApi\Model\FetchBatchesInterface;
+use Magento\Framework\App\ResourceConnection;
+use Magento\MediaContentApi\Api\DeleteContentAssetLinksByAssetIdsInterface;
+
+/**
+ * Remove obsolete content asset from deleted entities
+ */
+class RemoveObsoleteContentAsset
+{
+    private const MEDIA_CONTENT_ASSET_TABLE = 'media_content_asset';
+    private const DEFAULT_IDENTITY_FIELD = 'entity_id';
+
+    /**
+     * @var FetchBatchesInterface
+     */
+    private $fetchBatches;
+    
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var DeleteContentAssetLinksByAssetIdsInterface
+     */
+    private $deleteContentAssetLinksByAssetIds;
+
+    /**
+     * @var array
+     */
+    private $entities;
+
+    /**
+     * @var array $entityTableNames
+     */
+    private $entityTableNames;
+
+    /**
+     * @var array $identityFields
+     */
+    private $identityFields;
+    
+    /**
+     * @param DeleteContentAssetLinksByAssetIdsInterface $deleteContentAssetLinks
+     * @param FetchBatchesInterface $fetchBatches
+     * @param ResourceConnection $resourceConnection
+     * @param array $identityFields
+     * @param array $entities
+     * @param array $entityTableNames
+     */
+    public function __construct(
+        DeleteContentAssetLinksByAssetIdsInterface $deleteContentAssetLinks,
+        FetchBatchesInterface $fetchBatches,
+        ResourceConnection $resourceConnection,
+        array $identityFields = [],
+        array $entities = [],
+        array $entityTableNames = []
+    ) {
+        $this->deleteContentAssetLinksByAssetIds = $deleteContentAssetLinks;
+        $this->fetchBatches = $fetchBatches;
+        $this->resourceConnection = $resourceConnection;
+        $this->entities = $entities;
+        $this->identityFields = $identityFields;
+        $this->entityTableNames = $entityTableNames;
+    }
+
+    /**
+     * Remove media content if entity already deleted.
+     */
+    public function execute(): void
+    {
+        foreach ($this->entities as $entity) {
+            $columns = ['entity_id', 'entity_type', 'asset_id'];
+            foreach ($this->fetchBatches->execute(self::MEDIA_CONTENT_ASSET_TABLE, $columns) as $batch) {
```

Comment:
Is it possible to optimize this operation by building a select that is joining `media_content_asset` to the entity tables and selects the records that are missing a match in the entity table side?